pkg/controller/plan/kubevirt.go

```diff
@@ -1287,6 +1293,69 @@ func (r *KubeVirt) vmPreference(vm *plan.VMStatus) (virtualMachine *cnv.VirtualM
 	return
 }
 
+func (r *KubeVirt) setInstanceType(vm *plan.VMStatus, object *cnv.VirtualMachine) (err error) {
+	if r.Plan.Spec.InstanceType == "" {
+		return
+	}
+	intancetypeName, kind, err := r.getInstanceType(vm, r.Plan.Spec.InstanceType)
+	if err != nil {
+		return
+	}
+	object.Spec.Instancetype = &cnv.InstancetypeMatcher{Name: intancetypeName, Kind: kind}
+	return
+}
+
+// Attempt to find a suitable instance type
+func (r *KubeVirt) getInstanceType(vm *plan.VMStatus, instancetypeName string) (name, kind string, err error) {
```

Comment:
will `name` ever be different than `instancetypeName` (which should be `instanceTypeName`)? if not, then please drop `name`